File: src/main/ngapp/src/app/wsObjects/joinGame.ts
import {Message} from "./message";

export class JoinGame extends Message {
	private _localStorageUUID: string;
	private _playerUUID: string;
	private _playerID: number;
	private _playerName: string;
	
	constructor() {
		super();
		this.type = "JoinGame";
	}

	decodeResponse(msgRaw: any) {
		this._playerUUID = msgRaw.playerUUID;
		this._playerID = msgRaw.playerID;
		this._playerName = msgRaw.playerName;
	}

	get localStorageUUID(): string {
		return this._localStorageUUID;
	}
	
	get playerUUID(): string {
		return this._playerUUID;
	}
	
	get playerID(): number {
		return this._playerID;
	}
	
	get playerName(): string {
		return this._playerName;
	}
	
	set localStorageUUID(value: string) {
		this._localStorageUUID = value;
	}
	
	set playerUUID(value: string) {
		this._playerUUID = value;
	}
	
	set playerID(value: number) {
		this._playerID = value;
	}
	
	set playerName(value: string) {
		this._playerName = value;
	}
	
}

File: src/main/java/requests/spark/websockets/objects/messages/incoming/JoinGame.java
package requests.spark.websockets.objects.messages.incoming;

import game.Game;
import game.GameManager;
import game.actors.Player;
import org.apache.log4j.Logger;
import org.json.JSONObject;
import requests.spark.websockets.WebSocketManager;
import requests.spark.websockets.WebSocketSession;
import requests.spark.websockets.objects.Message;
import requests.spark.websockets.objects.MessageType;
import requests.spark.websockets.objects.messages.outgoing.NewPlayerJoined;

@MessageType("JoinGame")
public class JoinGame extends Message {
    private static Logger log = Logger.getLogger(JoinGame.class);

    private String localStorageUUID = null;

    public void process() {
        // Perform required action
        Game currentGame = GameManager.getInstance().getCurrentGame();

        Player player = null;
        if (localStorageUUID != null) {
            player = currentGame.findPlayer(localStorageUUID);
        }

        if (player == null) {
            player = currentGame.createPlayer();
        }

        // Send off push request to listeners
        NewPlayerJoined newPlayerJoined = Message.create(NewPlayerJoined.class);
        newPlayerJoined.newPlayer(player);
        newPlayerJoined.sendTo(Message.ALL_ADMINS);

        // Register the player session within the server
        WebSocketSession webSocketSession = new WebSocketSession()
                .session(getSession())
                .type(WebSocketSession.TYPE_PLAYER);
        WebSocketManager.getInstance().addSession(webSocketSession);

        // Add response values to this response
        addResponseData("playerUUID", player.getUuid());
        addResponseData("playerID", player.getId());

        // Send response
        handleResponse();
    }

    public void populate(JSONObject jsonObject) {
        if (jsonObject.has("_localStorageUUID") && !jsonObject.isNull("_localStorageUUID")) {
            localStorageUUID = jsonObject.getString("_localStorageUUID");
        }
    }
}
